test/common/config/http_subscription_impl_test.cc

```diff
@@ -42,6 +42,20 @@ TEST_F(HttpSubscriptionImplTest, BadJsonRecovery) {
   verifyStats(3, 1, 0, 1, 7148434200721666028);
 }
 
+TEST_F(HttpSubscriptionImplTest, ConfigNotModified) {
+  startSubscription({"cluster0", "cluster1"});
+  EXPECT_CALL(random_gen_, random()).WillOnce(Return(0));
+  EXPECT_CALL(*timer_, enableTimer(_));
+  EXPECT_CALL(callbacks_, onConfigUpdateFailed(_));
+  http_callbacks_->onFailure(Http::AsyncClient::FailureReason::Reset);
```

Comment:
If you consider my suggestions (if you think they are good 😄), here I think you can have the following expectations:

```c++
  verifyStats(1, 0, 0, 1, 0);
  timerTick();
  verifyStats(2, 0, 0, 1, 0);

  // accept and modify.
  deliverConfigUpdate({"cluster0", "cluster1"}, "0", true, true, "200");
  verifyStats(3, 1, 0, 1, 7148434200721666028);

  // accept and does not modify.
  deliverConfigUpdate({"cluster0", "cluster1"}, "0", true, false, "304");
  verifyStats(4, 1, 0, 1, 7148434200721666028);
``` 

And remove all below lines.